server/auth/types/openid/routes.ts

```diff
@@ -271,5 +284,116 @@ export class OpenIdAuthRoutes {
         });
       }
     );
+
+    // captureUrlFragment is the first route that will be invoked in the SP initiated login.
+    // This route will execute the captureUrlFragment.js script.
+    this.core.http.resources.register(
+      {
+        path: '/auth/openid/captureUrlFragment',
```

Comment:
I am trying to learn/understand the whole flow - where/when is this route invoked? I do not see that we provide this route to the IDP anywhere?

Reply:
I think the script `captureUrlFragment.js` stores the URL hash in the browser's local storage, and then it constructs a final URL for redirection by adding a query parameter (redirectHash) indicating whether the hash is present. It also appends the nextUrl from the query parameters. 

The script checks if the hash is present `(oidcHash !== "")` and sets redirectHash to `true`. It then constructs the final URL with the nextUrl and redirectHash query parameters and redirects the browser to that URL.

(cc @cwperks plz correct me if I'm wrong)

Reply:
That's correct. Thank you @RyanL1997.